### PR TITLE
doc: GitHub PRs workflow

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None

Describe changes:
- adds a documentation about GitHub PRs workflow

Draft : to be discussed, and to see if the approved filter with `gh` command is good enough ?

Modifies https://github.com/OISF/suricata/pull/8925 by taking review into account

### CODE REVIEW

File: doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+A draft may be closed if it has not been updated in two months.
+
+Mergeable PRs
+~~~~~~~~~~~~~
+
+When a Pull Request is intended to be merged as is, the workflow is the following:
+ 1. get reviewed, and either request changes or get approved
+ 2. if approved, get staged in a next branch (with other PRs), wait for CI validation
+   (and eventually request changes if CI finds anything)
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:21]:
```suggestion
    (and eventually request changes if CI finds anything)
```

Comment (jufajardini) [reply doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:21]: sorry about that :x

File: doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
+GitHub PRs workflow
+===================
+
+Draft PRs
+~~~~~~~~~
+
+A PR should be marked as `draft` if it is not intended to be merged as is,
+but is waiting for some sort of feedback.
+The author of the PR should explicit what kind of feedback is expected (CI/QA run, discussion on the code, etc...)
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:9]:
```suggestion
The author of the PR should explicit what kind of feedback is expected (CI/QA
run, discussion on the code, etc...)
```

File: doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+``is:pr is:open draft:false review:none sort:updated-asc no:assignee``
+The whole team is responsible to assign a PR to someone precise within 2 weeks.
+
+When someone gets assigned a PR, the PR should get a review status within 2 weeks:
+either changes requested, approved, or assigned to someone else if more expertise is needed.
+
+GitHub filter for changes-requested PRs is ``is:pr is:open draft:false sort:updated-asc review:changes-requested``
+
+Such a PR may be closed if it has not been updated in two months.
+It is expected that the author creates a new PR with a new version of the patch as described in :ref:`Pull Requests Criteria <pull-requests-criteria>`.
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:34]:
```suggestion
It is expected that the author creates a new PR with a new version of the patch
as described in :ref:`Pull Requests Criteria <pull-requests-criteria>`.
```

File: doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#TODO filter does not work cf https://github.com/OISF/suricata/pull/8898
+
+GitHub filter for approved PRs that should go in a next branch is ``is:pr is:open draft:false sort:updated-asc review:approved``
+
+Command to get approved PRs is ``gh pr list --json number,reviewDecision --search "state:open type:pr -review:none" | jq '.[] | select(.reviewDecision=="")'``
+
+Once in approved state, the PRs are in the responsibility of the merger, along with the next branches/PRs.
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:42]:
```suggestion
Once in approved state, the PRs are in the responsibility of the merger, along
with the next branches/PRs.
```

File: doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ 3. get merged and closed
+
+A newly created PR should match the filter
+``is:pr is:open draft:false review:none sort:updated-asc no:assignee``
+The whole team is responsible to assign a PR to someone precise within 2 weeks.
+
+When someone gets assigned a PR, the PR should get a review status within 2 weeks:
+either changes requested, approved, or assigned to someone else if more expertise is needed.
+
+GitHub filter for changes-requested PRs is ``is:pr is:open draft:false sort:updated-asc review:changes-requested``
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:31]:
```suggestion
GitHub filter for changes-requested PRs is ``is:pr is:open draft:false sort:
updated-asc review:changes-requested``
```

File: doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ 2. if approved, get staged in a next branch (with other PRs), wait for CI validation
+   (and eventually request changes if CI finds anything)
+ 3. get merged and closed
+
+A newly created PR should match the filter
+``is:pr is:open draft:false review:none sort:updated-asc no:assignee``
+The whole team is responsible to assign a PR to someone precise within 2 weeks.
+
+When someone gets assigned a PR, the PR should get a review status within 2 weeks:
+either changes requested, approved, or assigned to someone else if more expertise is needed.
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:29]:
```suggestion
either changes requested, approved, or assigned to someone else if more
expertise is needed.
```

File: doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+GitHub filter for changes-requested PRs is ``is:pr is:open draft:false sort:updated-asc review:changes-requested``
+
+Such a PR may be closed if it has not been updated in two months.
+It is expected that the author creates a new PR with a new version of the patch as described in :ref:`Pull Requests Criteria <pull-requests-criteria>`.
+
+#TODO filter does not work cf https://github.com/OISF/suricata/pull/8898
+
+GitHub filter for approved PRs that should go in a next branch is ``is:pr is:open draft:false sort:updated-asc review:approved``
```
Comment (jufajardini) [inline doc/userguide/devguide/codebase/contributing/github-pr-workflow.rst:38]:
```suggestion
GitHub filter for approved PRs that should go in a next branch is ``is:pr is:open
draft:false sort:updated-asc review:approved``
```

**Review by jufajardini — CHANGES_REQUESTED** [review]
Ok, I ran one last picky review, and submitted everything as suggestions, so, if you want, we can try to commit these and see if this version can be merged (IMO).

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/8946

**Pull request closed without merging**